gtars/src/common/models/region_set.rs

```diff
@@ -84,30 +188,276 @@ impl<'a> IntoIterator for &'a RegionSet {
+
+        for region in &self.regions {
+            buffer.push_str(&format!(
+                "{}\t{}\t{}\t{}\n",
+                region.chr,
+                region.start,
+                region.end,
+                region.rest.as_deref().unwrap_or("")
+            ));
+        }
+
+        let mut encoder = GzEncoder::new(BufWriter::new(file), Compression::fast());
+        encoder.write_all(buffer.as_bytes())?;
+
         Ok(())
     }
 
+    ///
+    /// Calculate identifier for RegionSet
+    ///
+    /// This function doesn't sort file, and identifer is based on
+    /// unsorted first 3 columns.
+    ///
+    /// # Returns
+    /// String containing RegionSet identifier
+    pub fn identifier(&self) -> String {
+        let mut chrs = String::new();
+        let mut starts = String::new();
+        let mut ends = String::new();
+
+        let mut first = true;
+        for region in &self.regions {
+            if !first {
+                chrs.push(',');
+                starts.push(',');
+                ends.push(',');
+            }
+            chrs.push_str(&region.chr);
+            starts.push_str(&region.start.to_string());
+            ends.push_str(&region.end.to_string());
+
+            first = false;
+        }
+
+        // Update hasher with input data
+
+        let mut hasher = Md5::new();
+        hasher.update(chrs);
+        let chrom_hash = hasher.finalize();
+
+        let mut hasher = Md5::new();
+        hasher.update(starts);
+        let start_hash = hasher.finalize();
+
+        let mut hasher = Md5::new();
+        hasher.update(ends);
+        let end_hash = hasher.finalize();
+
+        let combined = format!("{:x},{:x},{:x}", chrom_hash, start_hash, end_hash);
+
+        let mut hasher = Md5::new();
+        hasher.update(combined);
+        let hash = hasher.finalize();
+        let bed_digest: String = format!("{:x}", hash);
+
+        bed_digest
+    }
+
+    ///
+    /// Save RegionSet as bigBed (binary version of bed file)
+    ///
+    /// # Arguments
+    /// - out_path: the path to the bigbed file which should be created
+    /// - chrom_size: the path to chrom sizes file
+    ///
+    pub fn to_bigbed<T: AsRef<Path>>(&self, out_path: &Path, chrom_size: T) -> Result<()> {
+        let chrom_sizes: HashMap<String, u32> = get_chrom_sizes(chrom_size);
+
+        let region_vector = self.regions.iter().map(|i| {
+            // This if is removing regions that are not in chrom sizes file.
+            if !chrom_sizes.contains_key(&i.chr) {
+                return None;
+            }
+            Some(Ok::<_, Error>((
+                i.chr.clone(),
+                BedEntry {
+                    start: i.start,
+                    end: i.end,
+                    rest: i.rest.as_deref().unwrap_or("").to_string(),
+                },
+            )))
+        });
+
+        let region_vector = region_vector.filter(|e| e.is_some()).map(|e| e.unwrap());
```

Comment:
`filter` for `Some` followed by `unwrap`. consider using `flatten` instead: `flatten()`

Reply:
Actually I think you wrote it haha

Reply:
Oh yeah I remember... I saw this `clippy` warning, but I didn't like how unclear it was, so I opted for whats here because its more obvious whats going on, imo

Reply:
Sure, I'll leave that up to you